src/Assignments/ProblemDetails.tsx

```diff
@@ -20,17 +20,40 @@ export const ProblemDetails: React.FC<ProblemDetailsProps> = ({
     topic,
     attemptsRemaining,
 }) => {
-    const isAssessment = !_.isNil(topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0]);
-    const isClosed = isAssessment && topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0].isClosed;
-    const versionStartTime = topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0].startTime;
-    const versionEndTime = topic?.topicAssessmentInfo?.studentTopicAssessmentInfo?.[0].endTime;
+    if (_.isNil(topic)) {
+        console.error('Problem details requested without a topic');
```

Comment:
Should we be returning here?

Reply:
we cannot - react expects this component to always return from the same place?

Reply:
You can return from other places, but react expects a jsx return
We could return something like:
```
return (<p>An error occurrend</p>)
```

You should also be able to return null or `(<></>)`

But it could also if we have an error handler we could throw an error, if it's further down you could move the try up